Dockerfile-e2e

```diff
@@ -5,8 +5,8 @@ USER root
 RUN microdnf install -y tar gzip make which gcc gcc-c++ cyrus-sasl-lib findutils git
 
 # install platform specific go version
-RUN curl -O -J  https://dl.google.com/go/go1.22.5.linux-${TARGETARCH}.tar.gz
-RUN tar -C /usr/local -xzf go1.22.5.linux-${TARGETARCH}.tar.gz
+RUN curl -O -J  https://dl.google.com/go/go1.23.6.linux-${TARGETARCH}.tar.gz
```

Comment:
Didnt realize e2e was still using upstream go. Possibly we should switch these out to go-toolset for simplicity? I dont think we have to force the FIPS build or anything since it won't run on  FIPS system but we would never have to change the GO version again which would save time in future? Not a big deal for now, but something we should consider

Reply:
Yeah good call out. I think that's the play, at least to keep things consistent and accurate in comparison to our live envs. I'll get a card in the backlog.

Reply:
Oh yeah I must've missed this when I was updating relations because that was exactly what I did there, https://github.com/project-kessel/relations-api/blob/main/api/Dockerfile makes things a lot simpler as well.